File: RuntimeSystem-Benchmarking/Code/Matrix_Multiplication.py
import random
# Define the size of n*n matrix
size = 10
matrix1 = [[0]* size] * size
matrix2 = [[0] * size] *size
res = [[0] * size ] * size
# Min and max value for random double generator
minval = 0
maxval = 10
# Generate matrix of random numbers
for i in range(size):
    for j in range(size):
        matrix1[i][j] = random.uniform(minval,maxval)
        matrix2[i][j] = random.uniform(minval, maxval)



# Matrix multiplication
for i in range(size):
    for j in range(size):
        for k in range(size):
            res[i][j] += matrix1[i][k] * matrix2[k][j]

print(res)

File: RuntimeSystem-Benchmarking/Code/Matrix_Multiplication.java
import java.util.Random;
public class MatrixJava {
    public static void main(String[] args) {
        // Size of the n*n matrix
        int size = 100;
        Random rd = new Random();
        // The max value for range of random double values
        double max = 100;
        // The min value for range of random double values
        double min = 0;
        double matrix1[][] = new double[size][size];
        double matrix2[][] = new double[size][size];
        double res[][] = new double[size][size];
        // Initializing matrix with random values
        for (int i=0; i < size; i++){
            for (int j = 0; j < size; j++){
                matrix1[i][j] = rd.nextDouble()*(max - min) + min;
                matrix2[i][j] = rd.nextDouble()*(max - min) + min;
            }
        }

        // Matrix multiplication
        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {
                res[i][j] = 0;
                for (int k = 0; k < size; k++) {
                    res[i][j] += matrix1[i][k] * matrix2[k][j];
                }//end of k loop
            }

        }
    }
}